cfgov/v1/management/commands/report_raw_html.py

```diff
@@ -0,0 +1,142 @@
+        for index, item in enumerate(data):
+            local_path = path + [str(index)]
+            yield from find_pattern(item, pattern, key, path=local_path)
+
+
+class Command(BaseCommand):
+    help = (
+        "Discover raw HTML tags within &lt; &gt; entities in page fields. "
+        "Pass a list in the form of app_name.page_type.field for each page "
+        "type and field you want to report on."
+    )
+
+    # Match "&lt;" followed by any 0 or more characters that are not "&gt;",
+    # followed by "&gt;" This should match HTML that's encoded in HTML entities
+    # in the given fields, but it would be good to verify reported fields before
+    # assuming to much
+    html_tag_entity_pattern = r"&lt;[a-zA-Z]+(?!&gt;).*?&gt;"
+    html_tag_entity_re = re.compile(
+        html_tag_entity_pattern, re.MULTILINE | re.DOTALL
+    )
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "pagetype",
+            nargs="*",
+            help=(
+                "Specify the page type(s) and field to check."
+                "This should be given in the form app_name.page_type.field "
+                "to include a page type in the given app with the given field."
+                "For example, v1.BrowsePage.content. "
+                "If no page types are given, -f/--field is required."
+            ),
+        )
+
+    def handle(self, *args, **options):
+        raw_html_writer = csv.writer(self.stdout)
+        raw_html_writer.writerow(
+            (
+                "Page ID",
+                "Page Type",
+                "Page Title",
+                "Page URL",
+                "Raw HTML Block Path",
+                "Raw HTML",
+            )
+        )
+
+        pagetypes = options["pagetype"]
+        if len(pagetypes) == 0:
+            pagetypes = self.get_all_page_models_and_stream_fields()
+
+        for app_name_page_type_field in pagetypes:
+            app_name, page_type, field = app_name_page_type_field.split(".")
+            for match_data in self.get_matches(app_name, page_type, field):
+                raw_html_writer.writerow(match_data)
+
+    def get_all_page_models_and_stream_fields(self):
+        page_models = [
+            m for m in apps.get_models() if issubclass(m, CFGOVPage)
+        ]
+
+        pagetypes = []
+        for page_model in page_models:
+            for field in page_model._meta.fields:
+                if issubclass(field.__class__, StreamField):
+                    pagetypes.append(
+                        f"{page_model._meta.app_label}."
+                        f"{page_model._meta.object_name}."
+                        f"{field.name}"
+                    )
+
+        return pagetypes
+
+    def get_matches(self, app_name, page_type, field):
+        field_filter = {f"{field}__iregex": self.html_tag_entity_pattern}
+        PageModel = apps.get_model(app_label=app_name, model_name=page_type)
```

Comment:
One issue with iterating pagetype-by-pagetype is that it results in duplicate listings in the results. For example, LearnPage inherits from AbstractFilterPage, and so pages of this type are reported in multiple iterations of this method.

Would adding a call to [`exact_type`](https://docs.wagtail.org/en/stable/reference/pages/queryset_reference.html#wagtail.query.PageQuerySet.exact_type) somewhere here avoid this problem?

Reply:
Good call! Yes, and this reduces the number of matches by 175!

Reply:
Done in [649ac82](https://github.com/cfpb/consumerfinance.gov/pull/7981/commits/649ac82f8b65df6c70787f858441aa7f014fbfa4)